syntax = "proto3";

package average;

option go_package = "grpc-average/generated";

service Calculator {
  // Unary RPC: Computes the average of two numbers
  rpc Average (AverageRequest) returns (AverageResponse);

  // Server Streaming RPC: Computes the running average of a stream of numbers
  rpc RunningAverage (stream NumberRequest) returns (AverageResponse);
}

message AverageRequest {
  int32 num1 = 1;
  int32 num2 = 2;
}

message AverageResponse {
  double result = 1;
}

message NumberRequest {
  int32 number = 1;
}
